ui/pages/confirmations/confirm/confirm.tsx

```diff
@@ -1,61 +1,66 @@
+import { GasFeeContextProvider } from '../../../contexts/gasFee';
+import { TransactionModalContextProvider } from '../../../contexts/transaction-modal';
+import AdvancedGasFeePopover from '../components/advanced-gas-fee-popover';
+import { ConfirmAlerts } from '../components/confirm/confirm-alerts';
 import { Footer } from '../components/confirm/footer';
 import { Header } from '../components/confirm/header';
 import { Info } from '../components/confirm/info';
 import { LedgerInfo } from '../components/confirm/ledger-info';
 import { Nav } from '../components/confirm/nav';
 import { PluggableSection } from '../components/confirm/pluggable-section';
-import { TransactionModalContextProvider } from '../../../contexts/transaction-modal';
-import { GasFeeContextProvider } from '../../../contexts/gasFee';
-import AdvancedGasFeePopover from '../components/advanced-gas-fee-popover';
-import EditGasFeePopover from '../components/edit-gas-fee-popover';
-import { ConfirmAlerts } from '../components/confirm/confirm-alerts';
+import ScrollToBottom from '../components/confirm/scroll-to-bottom';
 import { Title } from '../components/confirm/title';
+import EditGasFeePopover from '../components/edit-gas-fee-popover';
+import setConfirmationAlerts from '../hooks/setConfirmationAlerts';
+import setCurrentConfirmation from '../hooks/setCurrentConfirmation';
+import syncConfirmPath from '../hooks/syncConfirmPath';
+import useConfirmationAlertActions from '../hooks/useConfirmationAlertActions';
 
 const Confirm = () => {
   const currentConfirmation = setCurrentConfirmation();
   syncConfirmPath();
+  setConfirmationAlerts();
+
+  const processAction = useConfirmationAlertActions();
 
   const [showAdvancedDetails, setShowAdvancedDetails] = useState(false);
 
   return (
-    <TransactionModalContextProvider>
-      <GasFeeContextProvider transaction={currentConfirmation}>
-        <EditGasFeePopover />
-        <AdvancedGasFeePopover />
-        <ConfirmAlerts>
-          <Page className="confirm_wrapper">
-            <Nav />
-            <Header
-              showAdvancedDetails={showAdvancedDetails}
-              setShowAdvancedDetails={setShowAdvancedDetails}
-            />
-            {
-              ///: BEGIN:ONLY_INCLUDE_IF(build-mmi)
-              <MMISignatureMismatchBanner />
-              ///: END:ONLY_INCLUDE_IF
-            }
-            <ScrollToBottom showAdvancedDetails={showAdvancedDetails}>
-              <LedgerInfo />
-              <Title />
-              <Info showAdvancedDetails={showAdvancedDetails} />
-              <PluggableSection />
-            </ScrollToBottom>
-            <Footer />
-          </Page>
-        </ConfirmAlerts>
-      </GasFeeContextProvider>
-    </TransactionModalContextProvider>
+    <AlertActionHandlerProvider onProcessAction={processAction}>
+      <TransactionModalContextProvider>
+        <GasFeeContextProvider transaction={currentConfirmation}>
```

Comment:
This is old context we should get rid of it.

Reply:
This is necessary for the current gas editing pop ups. I'm adding a comment inline to note this though.